ansible_base/authentication/authenticator_plugins/oidc.py

```diff
@@ -215,3 +217,49 @@ def extra_data(self, user, backend, response, *args, **kwargs):
 
     def get_user_groups(self, extra_groups=[]):
         return extra_groups
+
+    def public_key(self):
+        key = self.setting("PUBLIC_KEY")
+        if key:
+            head = "-----BEGIN PUBLIC KEY-----"
+            foot = "-----END PUBLIC KEY-----"
+            return (
+                "\n".join(
+                    [
+                        head,
+                        key,
+                        foot,
+                    ]
+                )
+                if head not in key
+                else key
+            )
+        return None
+
+    def user_data(self, access_token, *args, **kwargs):
+        """
+        This function overrides the one in social auth class OpenIdConnectAuth, since
+        it assumes the user info endpoint response returns plain json. Depending on
+        server config it may instead be a JWT. This function notices the JWT via the
+        content type header and if found, attempts to decode it using the configured
+        public key and algorithm(s). None is returned to signify a failed login in the
+        case of improper config or other decoding failure.
+        """
+        user_data = self.request(self.userinfo_url(), headers={"Authorization": f"Bearer {access_token}"})
+        if user_data.headers["Content-Type"] == "application/jwt":
```

Comment:
```suggestion
        if user_data.headers["Content-Type"] == "application/jwt":
            # If the content type is application/jwt than we can assume that the token is encrypted. Otherwise it should be application/json
```